plugins/content/bower/index.js

```diff
@@ -835,43 +835,70 @@ function handleUploadedPlugin (req, res, next) {
           return next(err);
         }
 
-        // first entry should be our target directory
         var packageJson;
-        var canonicalDir = path.join(outputPath, files[0]);
-        try {
-          packageJson = require(path.join(canonicalDir, 'bower.json'));
-        } catch (error) {
-          return next(error);
-        }
+        var canonicalDir;
+        // Read over each directory checking for the correct one that contains a bower.json file
+        fs.readdir(outputPath, function (err, directoryList) {
+
+          async.some(directoryList, function(directory, asyncCallback) {
+            var bowerPath = path.join(outputPath, directory, 'bower.json');
+
+            fs.exists(bowerPath, function(exists) {
+              if (exists) {
+                canonicalDir = path.join(outputPath, directory);
+                try {
+                  packageJson = require(bowerPath);
+                } catch (e) {
+                  return asyncCallback();
```

Comment:
yep - shame about that. In that case, maybe just logger.log the exception in the catch block, and pass a custom error to the callback on line 862.